File: 75.Sort Colors.py
class Solution(object):
    def sortColors(self, nums):
        """
        :type nums: List[int]
        :rtype: None Do not return anything, modify nums in-place instead.
        """
        sortedPtr = 0
        lookFor = 0
        lookForPtr = 0
        while sortedPtr < len(nums):
            while lookForPtr < len(nums) and nums[lookForPtr] != lookFor:
                lookForPtr += 1
            if lookForPtr == len(nums):
                lookFor += 1
                lookForPtr = sortedPtr
            else:
                nums[sortedPtr], nums[lookForPtr] = nums[lookForPtr], nums[sortedPtr]
                sortedPtr += 1
                lookForPtr += 1

File: 75.Sort Colors.rb
# @param {Integer[]} nums
# @return {Void} Do not return anything, modify nums in-place instead.
def sort_colors(nums)
    l, p, r = 0, 0, nums.size - 1
    while p <= r
        case nums[p]
        when 0
            nums[l], nums[p] = nums[p], nums[l]
            l += 1
            p = [l, p].max
        when 1
            p += 1
        when 2
            nums[r], nums[p] = nums[p], nums[r]
            r -= 1
        end
    end
end
